src/CommunityToolkit.Maui.MediaElement/Extensions/MetaDataExtensions.macios.cs

```diff
@@ -0,0 +1,180 @@
+	}
+
+	MPRemoteCommandHandlerStatus SeekForwardCommand(MPRemoteCommandEvent commandEvent)
+	{
+		if (player is not null)
+		{
+			var seekTime = player.CurrentTime + CMTime.FromSeconds(10, 1);
+			player.Seek(seekTime);
+		}
+		return MPRemoteCommandHandlerStatus.Success;
+	}
+
+	MPRemoteCommandHandlerStatus PlayCommand(MPRemoteCommandEvent commandEvent)
+	{
+		player?.Play();
+		return MPRemoteCommandHandlerStatus.Success;
+	}
+
+	MPRemoteCommandHandlerStatus PauseCommand(MPRemoteCommandEvent commandEvent)
+	{
+		player?.Pause();
+		return MPRemoteCommandHandlerStatus.Success;
+	}
+
+	MPRemoteCommandHandlerStatus ToggleCommand(MPRemoteCommandEvent commandEvent)
+	{
+		if (player?.Rate == 0)
+		{
+			player?.Play();
+		}
+		else
+		{
+			player?.Pause();
+		}
+		return MPRemoteCommandHandlerStatus.Success;
+	}
+
+	/// <summary>
+	/// Clears the metadata for the currently playing media.
+	/// </summary>
+	public void ClearNowPlaying()
+	{
+		NowPlayingInfo.AlbumTitle = string.Empty;
+		NowPlayingInfo.Title = string.Empty;
+		NowPlayingInfo.Artist = string.Empty;
+		NowPlayingInfo.AlbumTitle = string.Empty;
+		NowPlayingInfo.PlaybackDuration = 0;
+		NowPlayingInfo.IsLiveStream = false;
+		NowPlayingInfo.PlaybackRate = 0;
+		NowPlayingInfo.ElapsedPlaybackTime = 0;
+		NowPlayingInfo.Artwork = new MPMediaItemArtwork(new UIImage());
+		MPNowPlayingInfoCenter.DefaultCenter.NowPlaying = NowPlayingInfo;
+	}
+
+
+	/// <summary>
+	/// Sets the data for the currently playing media from the media element.
+	/// </summary>
+	/// <param name="PlayerItem"></param>
+	/// <param name="MediaElement"></param>
+	public void SetMetaData(AVPlayerItem? PlayerItem, IMediaElement MediaElement)
+	{
+		if (MediaElement is null)
+		{
+			ClearNowPlaying();
+			return;
+		}
+		NowPlayingInfo.Title = MediaElement.MetaDataTitle;
+		NowPlayingInfo.Artist = MediaElement.MetaDataArtist;
+		NowPlayingInfo.PlaybackDuration = PlayerItem?.Duration.Seconds ?? 0;
+		NowPlayingInfo.IsLiveStream = false;
+		NowPlayingInfo.PlaybackRate = MediaElement.Speed;
+		NowPlayingInfo.ElapsedPlaybackTime = PlayerItem?.CurrentTime.Seconds ?? 0;
+		NowPlayingInfo.Artwork = GetArtwork(MediaElement.MetaDataArtworkUrl);
+		MPNowPlayingInfoCenter.DefaultCenter.NowPlaying = NowPlayingInfo;
+	}
+
+	static MPMediaItemArtwork GetArtwork(string? ImageUri)
+	{
+		var defaultImage = new MPMediaItemArtwork(new UIImage());
+		try
+		{
+			if (!string.IsNullOrWhiteSpace(ImageUri))
+			{
+				UIImage? image = GetImage(ImageUri);
+				if (image is not null)
+				{
+					return new MPMediaItemArtwork(image);
+				}
+				return defaultImage;
+			}
+			return defaultImage;
+		}
+		catch
+		{
+			return defaultImage;
+		}
+		
+	}
+
+	static UIImage GetImage(string ImageUri)
+	{
+		return ImageUri switch
+		{
+			_ when ImageUri.StartsWith("http", StringComparison.CurrentCulture) 
+			=> (UIImage.LoadFromData(NSData.FromUrl(new NSUrl(ImageUri))) ?? new UIImage()),
+			_ => new UIImage()
```

Comment:
let's cache all these empty `UIImage`, that way we will not create a new one each time.

Reply:
do we need to return a new instance everytime?